HUMAN OVERSEER COMMENT - Parity: ThinkTool description matches Python SDK (#738)
enyst/OpenHands-Tab: packages/agent-sdk-ts/docs/python-parity.md

  5. Wrapping the returned result in an `ObservationEvent` where `observation` is a plain JSON object.
- Event interfaces in `src/sdk/types/index.ts` mirror the Python wire format: `ActionEvent` and `ObservationEvent` are present, but they carry raw records (`Record<string, unknown>`) instead of strongly-typed `Action`/`Observation` models.

**Built-in tools parity note (product decision / source of truth):** Python includes built-in `finish` and `think` tools by default. TypeScript includes both as built-ins in `src/sdk/runtime/Agent.ts` (unless `includeDefaultTools === false`).

> For `think`, **the TypeScript SDK MUST keep the tool description text identical to the Python SDK** (i.e., copy the Python `THINK_DESCRIPTION` verbatim). Even though the TS implementation is simpler, the tool description is part of the LLM-facing contract and directly affects agent behavior. We prioritize **cross-SDK behavioral parity** over minimizing prompt size for this specific tool.

Author: ```suggestion
For `think`, **the TypeScript SDK MUST keep the tool description text identical to the Python SDK** (i.e., copy the Python `THINK_DESCRIPTION` verbatim). Even though the TS implementation is simpler, the tool description is part of the LLM-facing contract and directly affects agent behavior. We prioritize **cross-SDK behavioral parity** over minimizing prompt size for all common tools.
```

> If any future optimization is desired (e.g., shorter descriptions to reduce prompt tokens), it must be treated as an explicit product decision and applied consistently across Python and TS (not a TS-only divergence).

Author: ```suggestion
If any future optimization is desired (e.g., shorter descriptions to reduce prompt tokens), it must be treated as an explicit product decision, applied consistently across Python and TS (not a TS-only divergence) and approved by Mail by the Human Overseer.
```
